lib/Runtime/HostManager/HostManager.cpp

```diff
@@ -112,17 +112,18 @@ llvm::Error HostManager::addNetwork(std::unique_ptr<Module> module,
   return llvm::Error::success();
 }
 
-void HostManager::removeNetwork(llvm::StringRef networkName) {
+llvm::Error HostManager::removeNetwork(llvm::StringRef networkName) {
   std::lock_guard<std::mutex> networkLock(networkLock_);
   auto networkIterator = networks_.find(networkName);
   if (networkIterator == networks_.end()) {
-    return;
+    return llvm::Error::success();
   }
 
-  // TODO: should return an error here rather than asserting:
-  // https://github.com/pytorch/glow/issues/2855
-  assert(networkIterator->second.refcount == 0 &&
-         "Cannot remove a network since there are outstanding runs");
+  // Issue an error as there are outstanding runs for the network
+  if (networkIterator->second.refcount != 0) {
+    return MAKE_ERR(GlowErr::ErrorCode::RUNTIME_REQUEST_REFUSED,
```

Comment:
RUNTIME_REQUEST_REFUSED is for inference requests that are refused, please find or create a more suitable error code for this case